tests/unittests/test_util.py

```diff
@@ -967,96 +967,125 @@ def test_get_proc_ppid(self):
-            return msdos
-        elif pattern == "/dev/iso9660/*":
-            return iso9660
-        raise Exception
-    m_glob.side_effect = fake_glob
-
-    devlist = util.find_devs_with_freebsd()
-    assert set(devlist) == set([
-        '/dev/iso9660/config-2', '/dev/msdosfs/EFISYS'])
-    devlist = util.find_devs_with_freebsd(criteria="TYPE=iso9660")
-    assert devlist == ['/dev/iso9660/config-2']
-    devlist = util.find_devs_with_freebsd(criteria="TYPE=vfat")
-    assert devlist == ['/dev/msdosfs/EFISYS']
-
-
-@mock.patch("cloudinit.subp.subp")
-def test_find_devs_with_netbsd(m_subp):
-    side_effect_values = [
-        ("ld0 dk0 dk1 cd0", ""),
-        (
+class TestFindDevs:
+    @mock.patch('cloudinit.subp.subp')
+    def test_find_devs_with(self, m_subp):
+        m_subp.return_value = (
+            '/dev/sda1: UUID="some-uuid" TYPE="ext4" PARTUUID="some-partid"',
+            ''
+        )
+        devlist = util.find_devs_with()
+        assert devlist == [
+            '/dev/sda1: UUID="some-uuid" TYPE="ext4" PARTUUID="some-partid"']
+
+        devlist = util.find_devs_with("LABEL_FATBOOT=A_LABEL")
+        assert devlist == [
+            '/dev/sda1: UUID="some-uuid" TYPE="ext4" PARTUUID="some-partid"']
+
+    @mock.patch('cloudinit.subp.subp')
+    def test_find_devs_with_openbsd(self, m_subp):
+        m_subp.return_value = (
+            'cd0:,sd0:630d98d32b5d3759,sd1:,fd0:', ''
+        )
+        devlist = util.find_devs_with_openbsd()
+        assert devlist == ['/dev/cd0a', '/dev/sd1i']
+
+    @mock.patch('cloudinit.subp.subp')
+    def test_find_devs_with_openbsd_with_criteria(self, m_subp):
+        m_subp.return_value = (
+            'cd0:,sd0:630d98d32b5d3759,sd1:,fd0:', ''
+        )
+        devlist = util.find_devs_with_openbsd(criteria="TYPE=iso9660")
+        assert devlist == ['/dev/cd0a']
+
+        # lp: #1841466
+        devlist = util.find_devs_with_openbsd(criteria="LABEL_FATBOOT=A_LABEL")
+        assert devlist == ['/dev/cd0a', '/dev/sd1i']
+
+    @mock.patch('glob.glob')
+    def test_find_devs_with_freebsd(self, m_glob):
+        def fake_glob(pattern):
+            msdos = ["/dev/msdosfs/EFISYS"]
+            iso9660 = ["/dev/iso9660/config-2"]
+            if pattern == "/dev/msdosfs/*":
+                return msdos
+            elif pattern == "/dev/iso9660/*":
+                return iso9660
+            raise Exception
+        m_glob.side_effect = fake_glob
+
+        devlist = util.find_devs_with_freebsd()
+        assert set(devlist) == set([
+            '/dev/iso9660/config-2', '/dev/msdosfs/EFISYS'])
+
+        devlist = util.find_devs_with_freebsd(criteria="TYPE=iso9660")
+        assert devlist == ['/dev/iso9660/config-2']
+
+        devlist = util.find_devs_with_freebsd(criteria="TYPE=vfat")
+        assert devlist == ['/dev/msdosfs/EFISYS']
+
+        # lp: #1841466
+        devlist = util.find_devs_with_freebsd(criteria="LABEL_FATBOOT=A_LABEL")
+        assert devlist == []
```

Comment:
has anyone tried booting a freebsd image under these conditions to see what the result looks like, and what we could be looking for?

Reply:
@igalic Was that one of the BSD people you pinged on the last ticket? Should I ping them here too? If we make an effort to reach out but they're unavailable, do you think the unit test changes are sufficient?

Reply:
We have roughly 3 *BSD people currently active:

- @goneri
- @ad-m
- @igalic

of those, @goneri is the most experienced when it comes to NoCloud

Reply:
Aloha,

Thank you @TheRealFalcon and @igalic, I will give a try to the patch.